### ISSUE TITLE
2.0.2 come across breakchange 

### ISSUE THREAD

**Issue by ngot** [socketio/socket.io#2963]
*Note*: for support questions, please use one of these channels: [stackoverflow](http://stackoverflow.com/questions/tagged/socket.io) or [slack](https://socketio.slack.com)

### You want to:

* [x] report a *bug*

### Current behaviour

```
/root/projects/agent-server/node_modules/_egg@1.3.0@egg/lib/application.js:38
      throw e;
      ^

TypeError: Cannot convert undefined or null to object
    at Namespace.use (/root/projects/agent-server/node_modules/_socket.io@2.0.2@socket.io/lib/namespace.js:103:21)
    at initNsp (/root/projects/agent-server/node_modules/_egg-socket.io@2.0.0@egg-socket.io/lib/io.js:80:9)
    at module.exports.app (/root/projects/agent-server/node_modules/_egg-socket.io@2.0.0@egg-socket.io/lib/io.js:55:5)
    at AppWorkerLoader.loadFile (/root/projects/agent-server/node_modules/_egg-core@3.7.0@egg-core/lib/loader/egg_loader.js:270:34)
    at getLoadUnits.forEach.unit (/root/projects/agent-server/node_modules/_egg-core@3.7.0@egg-core/lib/loader/mixin/custom.js:25:29)
    at Array.forEach (native)
    at AppWorkerLoader.loadCustomApp (/root/projects/agent-server/node_modules/_egg-core@3.7.0@egg-core/lib/loader/mixin/custom.js:25:8)
    at AppWorkerLoader.load (/root/projects/agent-server/node_modules/_egg@1.3.0@egg/lib/loader/app_worker_loader.js:33:10)
    at new Application (/root/projects/agent-server/node_modules/_egg@1.3.0@egg/lib/application.js:34:19)
    at Object.<anonymous> (/root/projects/agent-server/node_modules/_egg-cluster@1.6.2@egg-cluster/lib/app_worker.js:12:13)

```

### Steps to reproduce (if the current behaviour is a bug)

https://github.com/eggjs/egg-socket.io/tree/master/example run this demo

**Note**: the best way to get a quick answer is to provide a failing test case, by forking the following [fiddle](https://github.com/darrachequesne/socket.io-fiddle) for example.

### Expected behaviour

no error shows

### Setup
- OS: osx
- browser: 
- socket.io version: 2.0.2

### Other information (e.g. stacktraces, related issues, suggestions how to fix)
I have check this commit : 
https://github.com/socketio/socket.io/commit/2b216902e121ac2205444019b6d9316654809b29

if I delete this code .It will run ok. 

```js
delete this.server.eio.initialPacket;
```

It can't work with this commit, because:  `this.server.eio` is undefined!

So, what should i do with this issue?

**Comment by samypr100** [comment]
I'm getting the exactly the same issue 👍 
When I install `2.0.2`:
```
TypeError: Cannot convert undefined or null to object
    at Namespace.use (/path_to_project/socket.io/lib/namespace.js:103:21)
    at Server.(anonymous function) [as use] (/path_to_project/node_modules/socket.io/lib/index.js:443:29)
```
When I install `2.0.1`, everything works as usual.

**Comment by ngot** [comment]
anyone solve this？ or i just lock the version？

**Comment by samypr100** [comment]
I locked the version to `2.0.1` for now until this is fixed. I hope this gets someone's attention soon.

**Comment by darrachequesne** [comment]
Thanks, I could reproduce the issue, I'll try to fix it.

**Comment by ngot** [comment]
@darrachequesne What is the plan to fix this. Still broken.

**Comment by atian25** [comment]
any progress ?

**Comment by liutian** [comment]
I'm getting the exactly the same issue

**Comment by darrachequesne** [comment]
Fix released in [2.0.3](https://github.com/socketio/socket.io/releases/tag/2.0.3).
